frontend/src/components/management/WorkflowAssignment.tsx

```diff
@@ -0,0 +1,448 @@
+        const commonProjects = (await fetchProjects(projectIDs)).map((project) => ({
+            id: project.id,
+            name: project.name
+        }))
+        // console.info({ projectsBySample, countByProjectID, commonProjects })
+        setCommonProjects(commonProjects)
+
+        setIsFetching("Loading studies and workflows...")
+        const studyByID = projectIDs.length === 0 ? [] : (await dispatch(api.studies.list({ project__id__in: projectIDs.join(',') }))).data.results.reduce<Record<FMSStudy['id'], FMSStudy>>((acc, study) => {
+            acc[study.id] = study
+            return acc
+        }, {})
+        const workflowIDs = Object.values(studyByID).reduce<FMSWorkflow['id'][]>((acc, study) => {
+            acc.push(study.workflow_id)
+            return acc
+        }, [])
+        const workflowByID = (await fetchWorkflows(workflowIDs)).reduce<Record<FMSWorkflow['id'], FMSWorkflow>>((acc, workflow) => {
+            acc[workflow.id] = workflow
+            return acc
+        }, {})
+        // console.info({ studyByID, workflowByID })
+
+        setDequeueActions(
+            commonStudySteps.reduce<ActionInfo[]>((dequeueActions, [studyID, stepOrder]) => {
+                const workflow = workflowByID[studyByID[studyID].workflow_id]
+                const stepName = workflow.steps_order.find(s => s.order === stepOrder)?.step_name
+                if (stepName === undefined) return dequeueActions
+                dequeueActions.push({
+                    study: studyByID[studyID],
+                    workflow: workflow.name,
+                    step: stepName,
+                    stepOrder: stepOrder,
+                    alreadyQueued: [],
+                })
+                return dequeueActions
+            }, [])
+        )
+
+        const studyWorkflowsByProject: Record<Project['id'], StudyWorkflow[]> = {}
+        for (const project of commonProjects) {
+            studyWorkflowsByProject[project.id] = []
+        }
+
+        setQueueActions(Object.values(studyByID).reduce<ActionInfo[]>((queueActions, study) => {
+            if (!commonProjects.find(project => project.id === study.project_id)) return queueActions
+            const workflow = workflowByID[study.workflow_id]
+
+            const studyWorkflows = studyWorkflowsByProject[study.project_id] ?? []
+            studyWorkflows.push({ study: study.letter, workflow: workflow.name })
+            studyWorkflowsByProject[study.project_id] = studyWorkflows
+
+            for (const stepOrder of workflow.steps_order) {
+                if (stepOrder.order < study.start || stepOrder.order > study.end) continue
+                queueActions.push({
+                    study: study,
+                    workflow: workflow.name,
+                    step: stepOrder.step_name,
+                    stepOrder: stepOrder.order,
+                    alreadyQueued: [...sampleIDsByStudyStep[`${study.id}-${stepOrder.step_name}`] ?? []]
```

Comment:
You changed the key when creating the object this will likely never match anything.